addons/mail/static/src/models/thread/thread.js

```diff
@@ -645,36 +610,80 @@ registerModel({
             });
         },
         /**
-         * Fetches suggested recipients.
+         * Requests the given `requestList` data from the server.
+         *
+         * @param {string[]} requestList
          */
-        async fetchAndUpdateSuggestedRecipients() {
+        async fetchData(requestList) {
             if (this.isTemporary) {
                 return;
             }
-            return this.messaging.models['Thread'].performRpcMailGetSuggestedRecipients({
-                model: this.model,
-                res_ids: [this.id],
-            });
-        },
-        /**
-         * Fetch attachments linked to a record. Useful for populating the store
-         * with these attachments, which are used by attachment box in the chatter.
-         */
-        async fetchAttachments() {
-            const attachmentsData = await this.async(() => this.env.services.rpc({
-                model: 'ir.attachment',
-                method: 'search_read',
-                domain: [
-                    ['res_id', '=', this.id],
-                    ['res_model', '=', this.model],
-                ],
-                fields: ['id', 'name', 'mimetype'],
-                orderBy: [{ name: 'id', asc: false }],
-            }, { shadow: true }));
-            this.update({
-                originThreadAttachments: insertAndReplace(attachmentsData),
-            });
-            this.update({ areAttachmentsLoaded: true });
+            const requestSet = new Set(requestList);
+            if (!this.hasActivities) {
+                requestSet.delete('activities');
+            }
+            if (requestSet.has('attachments')) {
+                this.update({ isLoadingAttachments: true });
+            }
+            if (requestSet.has('messages')) {
+                this.cache.loadNewMessages();
+            }
```

Comment:
Can't this be included in batch? Or there is some reasons to intentionally make another RPC in parallel?

Reply:
1. I don't want to rewrite all thread cache because of this change.
2. It might be desirable to keep a decent level of split, to avoid one big RPC like init messaging. In particular it might be desirable to see attachments (example: invoice) and/or messages independently from each other as quickly as possible for each of them.

Reply:
Maybe some day (I would imagine having a single entry point that could allow us to batch any request, without us having to worry about it in code), but it is less priority than the rest of this PR.